From 73bb2317d38e807b94ac1e9993a0f1fe25b564af Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Thu, 6 Nov 2025 11:16:43 +0100
Subject: [PATCH] arm64: dts: st: align a35ss_syscfg node to the new bindings.

Fix base address and address range of a35ss_syscfg.
Adjust the address of the register in the client nodes.
Drop the extra compatible 'syscon'.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Ic6609c829abcb0ca853861fe149debd0482e94a9
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/509411
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi     | 6 +++---
 arch/arm64/boot/dts/st/stm32mp215f-dk.dts  | 2 +-
 arch/arm64/boot/dts/st/stm32mp231.dtsi     | 6 +++---
 arch/arm64/boot/dts/st/stm32mp235f-dk.dts  | 2 +-
 arch/arm64/boot/dts/st/stm32mp251.dtsi     | 6 +++---
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts  | 2 +-
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 2 +-
 7 files changed, 13 insertions(+), 13 deletions(-)

--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -2473,9 +2473,9 @@
 			};
 		};
 
-		a35ss_syscfg: syscon@48802000  {
-			compatible = "st,stm32mp25-a35ss-syscfg", "syscon";
-			reg = <0x48802000 0x0 0xac>;
+		a35ss_syscfg: syscon@48800000  {
+			compatible = "st,stm32mp25-a35ss-syscfg";
+			reg = <0x48800000 0x0 0x4000>;
 			status = "disabled";
 		};
 	};
--- a/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
@@ -568,7 +568,7 @@
 	memory-region = <&cm33_cube_fw>, <&cm33_cube_data>,
 			<&ipc_shmem_1>, <&vdev0vring0>,
 			<&vdev0vring1>, <&vdev0buffer>;
-	st,syscfg-nsvtor = <&a35ss_syscfg 0xa8 0xffffff80>;
+	st,syscfg-nsvtor = <&a35ss_syscfg 0x20a8 0xffffff80>;
 	status = "okay";
 };
 
--- a/arch/arm64/boot/dts/st/stm32mp231.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp231.dtsi
@@ -2736,9 +2736,9 @@
 			};
 		};
 
-		a35ss_syscfg: syscon@48802000  {
-			compatible = "st,stm32mp25-a35ss-syscfg", "syscon";
-			reg = <0x48802000 0xac>;
+		a35ss_syscfg: syscon@48800000  {
+			compatible = "st,stm32mp25-a35ss-syscfg";
+			reg = <0x48800000 0x4000>;
 			status = "disabled";
 		};
 
--- a/arch/arm64/boot/dts/st/stm32mp235f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp235f-dk.dts
@@ -528,7 +528,7 @@
 			<&ipc_shmem_1>, <&vdev0vring0>,
 			<&vdev0vring1>, <&vdev0buffer>,
 			<&cm33_sram2>;
-	st,syscfg-nsvtor = <&a35ss_syscfg 0xa8 0xffffff80>;
+	st,syscfg-nsvtor = <&a35ss_syscfg 0x20a8 0xffffff80>;
 	status = "okay";
 };
 
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -3109,9 +3109,9 @@
 			};
 		};
 
-		a35ss_syscfg: syscon@48802000  {
-			compatible = "st,stm32mp25-a35ss-syscfg", "syscon";
-			reg = <0x48802000 0xac>;
+		a35ss_syscfg: syscon@48800000  {
+			compatible = "st,stm32mp25-a35ss-syscfg";
+			reg = <0x48800000 0x4000>;
 			status = "disabled";
 		};
 
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -602,7 +602,7 @@
 			<&ipc_shmem_1>, <&vdev0vring0>,
 			<&vdev0vring1>, <&vdev0buffer>,
 			<&cm33_sram2>;
-	st,syscfg-nsvtor = <&a35ss_syscfg 0xa8 0xffffff80>;
+	st,syscfg-nsvtor = <&a35ss_syscfg 0x20a8 0xffffff80>;
 	status = "okay";
 };
 
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -646,7 +646,7 @@
 			<&ipc_shmem_1>, <&vdev0vring0>,
 			<&vdev0vring1>, <&vdev0buffer>,
 			<&cm33_sram2>;
-	st,syscfg-nsvtor = <&a35ss_syscfg 0xa8 0xffffff80>;
+	st,syscfg-nsvtor = <&a35ss_syscfg 0x20a8 0xffffff80>;
 	status = "okay";
 };
 
